### PR TITLE
Multiple tweaks to the Todo Overlay

### PR DESCRIPTION
**Pull request by Plantterror**
- Fix experiments resetting at midnight, when it should be at catacombs.
- Remove checking Cookie buff from player list, instead delegating that to opening the Skyblock Menu for a more accurate timer. I cannot do this with the God Potion due to false positives.
- Daily NPC Buy Limit, resets at catacombs. When you buy 640 (6400 during Diaz) of an item, this will activate the timer for the next day. 
(Right now this is based off of the chat message, I theorize it would take too many resources looping through all NPC shop names and shop items to check for the lore tag that is added, but it means that you can sometimes not activate the timer.)
- Crimson Isle Quests actually shows how many quests are needed to be completed.
(Might do this with heavy pearls, but this might take some adjustments to the current code)

Let me know if there's anything I could do better about this code or if you want me to add some other potential features while I'm at it.

### CODE REVIEW

**Comment by Plantterror** [comment]
Some more things i might do:
- Beacon Powering (suggested by Obsidian#5955)
- Cake timer updates while you attempt to eat a cake already consumed prior
- Daily dungeon runs? that code might be all over the place

**Review by romangraef — APPROVED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/overlays/TimersOverlay.java [hunk 1]
```diff
@@ -251,55 +261,140 @@ public void update() {
...
-				for (int i = 9; i < 18; i++) {
-					ItemStack stack = lower.getStackInSlot(i);
-					if (stack != null && stack.hasTagCompound()) {
-						String[] lore = NotEnoughUpdates.INSTANCE.manager.getLoreFromNBT(stack.getTagCompound());
-						for (String line : lore) {
-							if (line.contains("(Daily")) {
-								hidden.commissionsCompleted = 0;
-								break;
+			ItemStack stack = lower.getStackInSlot(0);
+			switch (containerName){
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/overlays/TimersOverlay.java:265]:
```suggestion
			switch (containerName.intern()) {
```

**Review by romangraef — CHANGES_REQUESTED** [review]
Also format

**Comment by Plantterror** [comment]
> Also format

Can you be more specific on this?

**Review by romangraef — APPROVED** [review]

**Comment by Plantterror** [comment]
I noticed a bug where the Election part of the player listing can actually trigger a crimson isle quest to be "completed" even though it shouldn't be. I'm going to work on that.

**Comment by Plantterror** [comment]
@jani270 This is both a bug fix and a feature drop, can you change the labels accordingly

**Review by romangraef — DISMISSED** [review]

**Review by romangraef — APPROVED** [review]

**Review by hannibal002 — APPROVED** [review]

**Review by Lulonaut — APPROVED** [review]

**Pull request merged**